ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -2296,10 +2305,9 @@ _WARN("Session space stuff is not implemented yet!")
         }
 
         /*
-         * Check if this is a real user-mode address or actually a kernel-mode
-         * page table for a user mode address
+         * Check if this is actually a user-mode address 
          */
-        if (Address <= MM_HIGHEST_USER_ADDRESS)
+        if (MiIsUserAddressOrPageTable(Address))
```

Comment:
This adds 2 additional checks for x86 (`MiIsUserPte` and `MiIsUserPde`)

Reply:
I removed the commit. Will open a separate PR for it later.